docs/docs/forms.mdx

```diff
@@ -32,13 +32,42 @@ The following example form `restaurant_form` will fill the slot
 ```yaml-rasa
 forms:
   restaurant_form:
-    cuisine:
-      - type: from_entity
-        entity: cuisine
-    num_people:
-      - type: from_entity
-        entity: number
+    global_not_intent: intent
+    required_slots:
+        cuisine:
+          - type: from_entity
+            entity: cuisine
+        num_people:
+          - type: from_entity
+            entity: number
+```
+
+There is also an optional parameter that can be defined in each form which is called
+`global_not_intent` parameter. It can be added under the form_name and its value will
+affect all slots that are present in that form. More specifically, it will update/add
+the `not_intent` key of all individual slots with the specified value (or list of values)
+found in the global parameter.
+
+For example, if you want to fill the `not_intent` key of all slots with `intent` then
+you would need to define the following after the form_name and before the definition
+of the slot mappings:
+
+```yaml-rasa
+forms:
+  restaurant_form:
+    global_not_intent: intent
+    required_slots:
+        cuisine:
+          - type: from_entity
+            entity: cuisine
+        num_people:
+          - type: from_entity
+            entity: number
 ```
+Note that if you decide to use the `global_not_intent` parameter then the keyword
+`required_slots` should precede the definition of slot mappings as shown in the above
```

Comment:
I would argue that we always support `required_slots` even if there is no `global_not_intent`. Maybe we also think about deprecating the lack of it? @akelad thoughts?

Reply:
I'm pretty sure that leaving required_slots empty is allowed if you have custom slot mappings? See point 2 here: https://rasa.com/docs/rasa/forms#custom-slot-mappings

Reply:
> I'm pretty sure that leaving required_slots empty is allowed if you have custom slot mappings?

True

> Maybe we also think about deprecating the lack of it

I would do this in case there are non custom slot mappings to push users to annotate them within `required_slots` and so that we can drop the extra code in 3.0

Reply:
I think the current implementation is fine, **but** I'd implement a deprecation warning for the case that users don't use `required_slots` to define slot mappings. I would always raise this warning independent if they use `global_not_intent` or not.

Means this causes a deprecation warning (but continues to work)

```yaml
forms:
  my_form:
    my_slot:
   - type: from_text
```

And these don't:
```yaml
forms:
  my_form:
    required_slots:
      my_slot:
     - type: from_text
```

```yaml
forms:
  my_form:
    global_not_intent:
    - intent1
    - intent2
    required_slots:
      my_slot:
     - type: from_text
```

> and in the function required_slots we define the ones with custom_mappings.

This function is in the SDK and doesn't need to be considered here, **but** it raises an important point:

* from what I understand `slot_mapping_for_form` breaks with this change as it won't return the slot mappings but rather `required_slots` 
* This in turns requires another change in the SDK [here](https://github.com/RasaHQ/rasa-sdk/blob/7f9d7a43a48c9fbec170453c8459c70648cfb8a8/rasa_sdk/forms.py#L820)